incubator/hnc/pkg/kubectl/set.go

```diff
@@ -16,77 +16,76 @@ limitations under the License.
 package kubectl
 
 import (
-	"errors"
 	"fmt"
 	"os"
+	"sort"
+	"strings"
 
 	"github.com/spf13/cobra"
 
 	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
 )
 
-//flagValues struct stores name of namespaces against type of flag passed
-type flagValues struct {
+//hcUpdates struct stores name of namespaces against type of flag passed
+type hcUpdates struct {
 	root             bool
 	parent           string
 	requiredChildren []string
-	optionalChild    string
+	optionalChildren []string
 }
 
 var setCmd = &cobra.Command{
-	Use:     "set",
-	Short:   "Allows setting namespace hierarchy",
-	Aliases: []string{"set"},
-	Args:    cobra.MinimumNArgs(1),
+	Use:   "set <namespace>",
+	Short: "Sets hierarchical properties of the given namespace",
+	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		nnm := args[0]
 		flags := cmd.Flags()
 		parent, _ := flags.GetString("parent")
 		requiredChildren, _ := flags.GetStringArray("requiredChild")
-		optionalChild, _ := flags.GetString("optionalChild")
+		optionalChildren, _ := flags.GetStringArray("optionalChild")
+		requiredChildren = normalizeStringArray(requiredChildren)
```

Comment:
Side question: If I'm reading this properly, this is an acceptable use case: `--requiredChild foo --requiredChild bar,baz` and turns into a single list. Any reason to not just use `setCmd.Flags().String` to have only common separated? Just curious if there is a particular use case I haven't thought of?

Reply:
Yup, that's exactly what we support. I've always _understood_ that some people prefer repeated flags while others prefer comma-delimited lists, so I just figured we'd support both. Nothing really deeper than that.

Reply:
Cool, that’s what I thought.